src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,555 @@
+    /// though ownerOf(id) would be address(0) after burning.)
+    /// @param id The id of the GAME token to query.
+    /// @return the address of the owner before burning.
+    function _withdrawalOwnerOf(uint256 id) internal view returns (address) {
+        uint256 data = _owners[_storageId(id)];
+        if ((data & (2**160)) == 2**160) {
+            return address(data);
+        }
+        return address(0);
+    }
+
+    /// @dev Create a new gameId and associate it with an owner.
+    /// @param from The address of one creating the game.
+    /// @param to The address of the Game owner.
+    /// @param subId The id to use when generating the new GameId.
+    /// @param version The version number part of the gameId.
+    /// @param isCreation Whether this is a brand new GAME (as opposed to an update).
+    /// @return id The newly created gameId.
+    function _mintGame(
+        address from,
+        address to,
+        uint64 subId,
+        uint32 version,
+        bool isCreation
+    ) internal returns (uint256 id, uint256 baseId) {
+        uint32 idVersion;
+        uint256 gameId;
+        uint256 storageId;
+        if (isCreation) {
+            idVersion = 1;
+            gameId = _generateGameId(from, subId, idVersion);
+            storageId = _storageId(gameId);
+            require(_owners[storageId] == 0, "BASEID_REUSE_FORBIDDEN");
+        } else {
+            idVersion = version;
+            gameId = _generateGameId(from, subId, idVersion);
+            storageId = _storageId(gameId);
+        }
+
+        _owners[storageId] = (uint256(idVersion) << 200) + uint256(to);
+
+        emit Transfer(address(0), to, gameId);
+        return (gameId, storageId);
+    }
+
+    /// @dev Allow token owner to set game editors.
+    /// @param gameCreator The address of a GAME creator,
+    /// @param editor The address of the editor to set.
+    /// @param isEditor Add or remove the ability to edit.
+    function _setGameEditor(
+        address gameCreator,
+        address editor,
+        bool isEditor
+    ) internal {
+        emit GameEditorSet(gameCreator, editor, isEditor);
+        _gameEditors[gameCreator][editor] = isEditor;
+    }
+
+    /// @dev Bumps the version number of a game token, buring the previous
+    /// version and minting a new one.
+    /// @param from The address of the GAME token owner.
+    /// @param gameId The Game token to bump the version of.
+    /// @return The new gameId.
+    function _bumpGameVersion(address from, uint256 gameId) internal returns (uint256) {
+        address originalCreator = address(gameId / CREATOR_OFFSET_MULTIPLIER);
+        uint64 subId = uint64(gameId / SUBID_MULTIPLIER);
+        uint32 version = uint32(gameId);
+        version++;
+        address owner = _ownerOf(gameId);
+        _burn(from, owner, gameId);
+        (uint256 newId, ) = _mintGame(originalCreator, owner, subId, version, false);
```

Comment:
_burn will call `_numNFTPerAddress[from]--`
but _mintGame will not call `_numNFTPerAddress[from]++`
so the balanceOf will be wrong

would be good idea to add test for taht

Reply:
good catch. I had moved the balance incrementer to `createGame` , but it doesn't accurately track balances in the context of immutable Game tokens. 
Fixing.